src/stactools/hotosm/cli.py

```diff
@@ -0,0 +1,180 @@
+"""Command line interface for OAM STAC creation and syncing."""
+
+import datetime as dt
+from typing import Any
+
+import click
+import pystac
+import requests
+from pypgstac.db import PgstacDB
+from pypgstac.load import Loader, Methods
+
+uploaded_since_sec = click.option(
+    "--uploaded-since",
+    type=float,
+    help="Find Items uploaded in last period provided in seconds.",
+)
+uploaded_after_dt = click.option(
+    "--uploaded-after",
+    type=click.DateTime(),
+    help="Find Items uploaded after this date (UTC).",
+)
+
+
+def parse_uploaded_since(
+    uploaded_since_sec: float | None,
+    uploaded_after_dt: dt.datetime | None,
+) -> dt.datetime:
+    """Parse mutually exclusive arguments for finding Items uploaded since some time."""
+    if (uploaded_after_dt is uploaded_since_sec) or (
+        uploaded_since_sec and uploaded_after_dt
+    ):
+        raise click.BadParameter(
+            "Must provide one of --uploaded-since or --uploaded-after"
+        )
+
+    if uploaded_since_sec is not None:
+        return dt.datetime.now(tz=dt.UTC) - dt.timedelta(seconds=uploaded_since_sec)
+
+    if uploaded_after_dt is not None:
+        return uploaded_after_dt.replace(tzinfo=dt.UTC)
+
+    raise click.BadParameter("Must provide one of --uploaded-since or --uploaded-after")
+
+
+def create_pgstac_from_ctx(
+    ctx: click.Context,
+    _: click.Parameter,
+    value: Any,
+) -> PgstacDB:
+    """Create PgSTAC connection."""
+    dsn = "postgresql://{user}:{password}@{host}:{port}/{database}".format(
+        user=ctx.params["pguser"],
+        password=ctx.params["pgpassword"],
+        host=ctx.params["pghost"],
+        port=ctx.params["pgport"],
+        database=value,
+    )
+    ctx.obj = ctx.obj or {}
+    ctx.obj["pgstac"] = PgstacDB(dsn)
+    return value
+
+
+pgstac_username = click.option(
+    "--pguser",
+    envvar="PGUSER",
+    help="Username for PgSTAC",
+    required=True,
+)
+pgstac_password = click.option(
```

Comment:
I had not seen this pattern (defining options first then using them as decorators) and I'll be using this, very nice.